server/src/main/java/org/elasticsearch/action/search/SearchProgressListener.java

```diff
@@ -225,7 +223,7 @@ static List<SearchShard> buildSearchShards(List<? extends SearchPhaseResult> res
             .toList();
     }
 
-    static List<SearchShard> buildSearchShards(GroupShardsIterator<SearchShardIterator> its) {
-        return StreamSupport.stream(its.spliterator(), false).map(e -> new SearchShard(e.getClusterAlias(), e.shardId())).toList();
+    static List<SearchShard> buildSearchShardsFromIter(List<SearchShardIterator> its) {
```

Comment:
super minor nit: we could have left the name as it was?

Reply:
Yea right this is annoying, unfortunately can't do that it conflicts with the other method of the same name. The existing other method is sort of single use only now, looks like we can clean this up in a follow-up actually.

Reply:
got it.